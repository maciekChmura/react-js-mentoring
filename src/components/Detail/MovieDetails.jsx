import React from 'react';
import MoviePoster from '../Body/MoviePoster/MoviePoster';
import {
  DetailsWrapper,
  InfoWrapper,
  MovieTitle,
  MovieRating,
  MovieData,
  MovieSummary
} from './MovieDetails.Styles';

const MovieDetails = ({ details }) => {
  const releaseDate = details.release_date.slice(0, 4);
  return (
    <DetailsWrapper>
      <MoviePoster posterImage={details.poster_path} />
      <InfoWrapper>
        <MovieTitle>{details.title}</MovieTitle>
        <MovieRating>Rating: {details.vote_average}</MovieRating>
        <MovieData>
          {releaseDate} - {details.runtime}min
        </MovieData>
        <MovieSummary> {details.overview} </MovieSummary>
      </InfoWrapper>
    </DetailsWrapper>
  );
};

export default MovieDetails;
