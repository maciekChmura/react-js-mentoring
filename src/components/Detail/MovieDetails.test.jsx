import React from 'react';
import { shallow } from 'enzyme';

import MovieDetails from './MovieDetails';

describe('MovieDetails', () => {
  const details = {
    id: 12,
    title: 'Finding Nemo',
    tagline:
      "There are 3.7 trillion fish in the ocean. They're looking for one.",
    vote_average: 7.7,
    vote_count: 8035,
    release_date: '2003-05-30',
    poster_path:
      'https://image.tmdb.org/t/p/w500/syPWyeeqzTQIxjIUaIFI7d0TyEY.jpg',
    overview:
      "Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.",
    budget: 94000000,
    revenue: 940335536,
    genres: ['Animation', 'Family'],
    runtime: 100
  };
  it('should render correctly', () => {
    const component = shallow(<MovieDetails details={details} />);

    expect(component).toMatchSnapshot();
  });
});
