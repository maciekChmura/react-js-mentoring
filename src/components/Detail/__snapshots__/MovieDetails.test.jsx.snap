// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieDetails
    details={
      Object {
        "budget": 94000000,
        "genres": Array [
          "Animation",
          "Family",
        ],
        "id": 12,
        "overview": "Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.",
        "poster_path": "https://image.tmdb.org/t/p/w500/syPWyeeqzTQIxjIUaIFI7d0TyEY.jpg",
        "release_date": "2003-05-30",
        "revenue": 940335536,
        "runtime": 100,
        "tagline": "There are 3.7 trillion fish in the ocean. They're looking for one.",
        "title": "Finding Nemo",
        "vote_average": 7.7,
        "vote_count": 8035,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <MoviePoster
          posterImage="https://image.tmdb.org/t/p/w500/syPWyeeqzTQIxjIUaIFI7d0TyEY.jpg"
        />,
        <ForwardRef>
          <ForwardRef>
            Finding Nemo
          </ForwardRef>
          <ForwardRef>
            Rating: 
            7.7
          </ForwardRef>
          <ForwardRef>
            2003
             - 
            100
            min
          </ForwardRef>
          <ForwardRef>
             
            Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.
             
          </ForwardRef>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "posterImage": "https://image.tmdb.org/t/p/w500/syPWyeeqzTQIxjIUaIFI7d0TyEY.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              Finding Nemo
            </ForwardRef>,
            <ForwardRef>
              Rating: 
              7.7
            </ForwardRef>,
            <ForwardRef>
              2003
               - 
              100
              min
            </ForwardRef>,
            <ForwardRef>
               
              Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.
               
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Finding Nemo",
            },
            "ref": null,
            "rendered": "Finding Nemo",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "MovieDetailsStyles__MovieTitle-ip2p7-2",
                "isStatic": true,
                "rules": Array [
                  "font-size:32px;color:#dd1133;",
                ],
              },
              "displayName": "MovieDetailsStyles__MovieTitle",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "MovieDetailsStyles__MovieTitle-ip2p7-2",
              "target": "p",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                "Rating: ",
                7.7,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Rating: ",
              7.7,
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "MovieDetailsStyles__MovieRating-ip2p7-3",
                "isStatic": true,
                "rules": Array [
                  "font-size:18px;color:white;",
                ],
              },
              "displayName": "MovieDetailsStyles__MovieRating",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "MovieDetailsStyles__MovieRating-ip2p7-3",
              "target": "p",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                "2003",
                " - ",
                100,
                "min",
              ],
            },
            "ref": null,
            "rendered": Array [
              "2003",
              " - ",
              100,
              "min",
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "MovieDetailsStyles__MovieData-ip2p7-4",
                "isStatic": true,
                "rules": Array [
                  "font-size:14px;color:white;",
                ],
              },
              "displayName": "MovieDetailsStyles__MovieData",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "MovieDetailsStyles__MovieData-ip2p7-4",
              "target": "p",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                " ",
                "Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.",
              " ",
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "MovieDetailsStyles__MovieSummary-ip2p7-5",
                "isStatic": true,
                "rules": Array [
                  "font-size:14px;color:white;",
                ],
              },
              "displayName": "MovieDetailsStyles__MovieSummary",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "MovieDetailsStyles__MovieSummary-ip2p7-5",
              "target": "p",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "MovieDetailsStyles__InfoWrapper-ip2p7-1",
            "isStatic": true,
            "rules": Array [
              "width:600px;margin:16px;",
            ],
          },
          "displayName": "MovieDetailsStyles__InfoWrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "MovieDetailsStyles__InfoWrapper-ip2p7-1",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "MovieDetailsStyles__DetailsWrapper-ip2p7-0",
        "isStatic": true,
        "rules": Array [
          "width:800px;height:400px;display:flex;",
        ],
      },
      "displayName": "MovieDetailsStyles__DetailsWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "MovieDetailsStyles__DetailsWrapper-ip2p7-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MoviePoster
            posterImage="https://image.tmdb.org/t/p/w500/syPWyeeqzTQIxjIUaIFI7d0TyEY.jpg"
          />,
          <ForwardRef>
            <ForwardRef>
              Finding Nemo
            </ForwardRef>
            <ForwardRef>
              Rating: 
              7.7
            </ForwardRef>
            <ForwardRef>
              2003
               - 
              100
              min
            </ForwardRef>
            <ForwardRef>
               
              Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.
               
            </ForwardRef>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "posterImage": "https://image.tmdb.org/t/p/w500/syPWyeeqzTQIxjIUaIFI7d0TyEY.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                Finding Nemo
              </ForwardRef>,
              <ForwardRef>
                Rating: 
                7.7
              </ForwardRef>,
              <ForwardRef>
                2003
                 - 
                100
                min
              </ForwardRef>,
              <ForwardRef>
                 
                Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.
                 
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Finding Nemo",
              },
              "ref": null,
              "rendered": "Finding Nemo",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "MovieDetailsStyles__MovieTitle-ip2p7-2",
                  "isStatic": true,
                  "rules": Array [
                    "font-size:32px;color:#dd1133;",
                  ],
                },
                "displayName": "MovieDetailsStyles__MovieTitle",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "MovieDetailsStyles__MovieTitle-ip2p7-2",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "Rating: ",
                  7.7,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Rating: ",
                7.7,
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "MovieDetailsStyles__MovieRating-ip2p7-3",
                  "isStatic": true,
                  "rules": Array [
                    "font-size:18px;color:white;",
                  ],
                },
                "displayName": "MovieDetailsStyles__MovieRating",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "MovieDetailsStyles__MovieRating-ip2p7-3",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "2003",
                  " - ",
                  100,
                  "min",
                ],
              },
              "ref": null,
              "rendered": Array [
                "2003",
                " - ",
                100,
                "min",
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "MovieDetailsStyles__MovieData-ip2p7-4",
                  "isStatic": true,
                  "rules": Array [
                    "font-size:14px;color:white;",
                  ],
                },
                "displayName": "MovieDetailsStyles__MovieData",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "MovieDetailsStyles__MovieData-ip2p7-4",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  " ",
                  "Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "Nemo, an adventurous young clownfish, is unexpectedly taken from his Great Barrier Reef home to a dentist's office aquarium. It's up to his worrisome father Marlin and a friendly but forgetful fish Dory to bring Nemo home -- meeting vegetarian sharks, surfer dude turtles, hypnotic jellyfish, hungry seagulls, and more along the way.",
                " ",
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "MovieDetailsStyles__MovieSummary-ip2p7-5",
                  "isStatic": true,
                  "rules": Array [
                    "font-size:14px;color:white;",
                  ],
                },
                "displayName": "MovieDetailsStyles__MovieSummary",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "MovieDetailsStyles__MovieSummary-ip2p7-5",
                "target": "p",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "MovieDetailsStyles__InfoWrapper-ip2p7-1",
              "isStatic": true,
              "rules": Array [
                "width:600px;margin:16px;",
              ],
            },
            "displayName": "MovieDetailsStyles__InfoWrapper",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "MovieDetailsStyles__InfoWrapper-ip2p7-1",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "MovieDetailsStyles__DetailsWrapper-ip2p7-0",
          "isStatic": true,
          "rules": Array [
            "width:800px;height:400px;display:flex;",
          ],
        },
        "displayName": "MovieDetailsStyles__DetailsWrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "MovieDetailsStyles__DetailsWrapper-ip2p7-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
